core/src/main/java/jenkins/monitor/JavaLevelAdminMonitor.java

```diff
@@ -0,0 +1,68 @@
+package jenkins.monitor;
+
+import hudson.Extension;
+import hudson.model.AdministrativeMonitor;
+import hudson.util.VersionNumber;
+import jenkins.model.Jenkins;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.DoNotUse;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.HttpRedirect;
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.HttpResponses;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+import java.io.IOException;
+
+@Extension
+@Restricted(NoExternalUse.class)
+public class JavaLevelAdminMonitor extends AdministrativeMonitor {
+
+    private static final VersionNumber RECOMMENDED_MINIMUM_VERSION = new VersionNumber("11");
+    private final VersionNumber currentJavaVersion;
+
+    @SuppressWarnings("unused")
+    public JavaLevelAdminMonitor() {
+        this(JavaLevelAdminMonitor.class.getName(), getJavaVersion());
+    }
+
+    public JavaLevelAdminMonitor(String id, VersionNumber currentJavaVersion) {
+        super(id);
+        this.currentJavaVersion = currentJavaVersion;
+    }
+
+    private static VersionNumber getJavaVersion() {
+        String version = System.getProperty("java.version");
```

Comment:
Should be `java.specification.version` I think, but no need to reinvent the wheel here, `jenkins.util.java.JavaUtils` probably has what you need.